### ISSUE TITLE
Add Cupertino equivalent of the ButtonBar

### ISSUE THREAD

**Issue by navaronbracke** [flutter/flutter#62286]
Recently I was figuring out how to roll my own version of a dialog widget.

For Android this was pretty straightforward and easy to implement.
Things like padding and text styles were well documented, so reusing those was easy.

However on IOS I ran into a bit of a problem.
The Material library has the `ButtonBar` widget, that is used by dialogs to display the dialog actions.
IOS on the other hand, doesn't have a (publicly available) equivalent.

The `_CupertinoAlertActionSection` StatefulWidget is private and thus we can't access it through the Cupertino library.

In short, I'd request a Cupertino equivalent of the `ButtonBar` widget, that can be used to create the dialog actions for IOS dialogs, no matter if the dialog you are creating is a `CupertinoAlertDialog` or a custom one with `CupertinoPopupSurface`.

**Comment by Piinks** [comment]
Hey @navaronbracke we just came across this issue in triage. ButtonBar is deprecated in favor of OverflowBar (in the widgets library), but we were wondering if there is a specific iOS component you were referring to here?

**Comment by navaronbracke** [comment]
I was not specifically referring to a component in the iOS library.

The main thing with the CupertinoAlertDialog's actions is how it handles the horizontal, thin border and the vertical border between buttons that are placed side-by-side. Back when I filed this issue, the widget that did that was part of the private API.

The closest thing that I _can_ refer to is the example of the dialog here: https://developer.apple.com/design/human-interface-guidelines/alerts#Anatomy

Although I think one could easily write their own version using OverflowBar and a bit of fiddling with the border lines around the buttons.

Feel free to close, I think the demand for this is quite low anyway.

**Comment by Piinks** [comment]
Thank you for the update!

> The main thing with the CupertinoAlertDialog's actions is how it handles the horizontal, thin border and the vertical border between buttons that are placed side-by-side.

I found https://github.com/flutter/flutter/issues/61164 that seems to be about this, so I think I will close this to refer to that issue. Thanks again!
